src/RoslynTestKit/AnalyzerTestFixture.cs

```diff
@@ -12,6 +12,17 @@ public abstract class AnalyzerTestFixture : BaseTestFixture
     {
         protected abstract DiagnosticAnalyzer CreateAnalyzer();
 
+        protected void NoException(string code)
+        {
+            var document = MarkupHelper.GetDocumentFromCode(code, LanguageName, References);
+            var diagnostics = GetDiagnostics(document);
+            var diagnostic = diagnostics.FirstOrDefault(d => d.Id == "AD0001");
+            if (diagnostic != null)
+            {
+                throw RoslynTestKitException.ExceptionInAnalyzer(diagnostic.Descriptor.Description);

```

Comment:
Not sure if we should use `Descriptor.Description` or rather `diagnostic.GetMessage(Thread.CurrentThread.CurrentCulture)`. Did you test the outcome?

Reply:
Did not know about this, thank you

Reply:
diagnostic.Descriptor.Description reports full stack trace, GetMessage reports just message without stack

Reply:
Description might be another property of exception, I think GetMessage is better for our case

Reply:
Maybe you should assemble the final message from both sources?

Reply:
You can create a specific type of exception that inherits from `RoslynTestKitException` and
-  accepts diagnostic as the parameter
- assign ` diagnostic.GetMessage(Thread.CurrentThread.CurrentCulture)` to the `Message` field, 
- override `ToString()` method and return message combined from `Message` and `Descriptor`

Reply:
@divan-9 can I merge it?

Reply:
I did try your suggestion, and seems like xunit does not call ToString()

## Reporting diagnostic.GetMessage
```
 X Sorry.Analyzers.Tests.Sorry1001Tests.DoesNotWarnEmptyArgumentList [1s 618ms]
  Error Message:
   RoslynTestKit.RoslynTestKitAnalyzerFailedException : Analyzer 'Sorry.Analyzers.Sorry1001WrongIndentation' threw an exception of type 'System.NotImplementedException' with message 'The method or operation is not implemented.'.
  Stack Trace:
     at RoslynTestKit.AnalyzerTestFixture.NoException(String code) in /Users/dmitryivanov/dev/sorry/deps/RoslynTestKit/src/RoslynTestKit/AnalyzerTestFixture.cs:line 24
   at Sorry.Analyzers.Tests.Sorry1001Tests.DoesNotWarnEmptyArgumentList() in /Users/dmitryivanov/dev/sorry/tests/Sorry.Analyzers.Tests/Sorry1001Tests.cs:line 24
                                                                                                                                              
Test Run Failed.
```

## Reporting diagnostic.Descriptor.Description
Note, it also contains original message
```
[xUnit.net 00:00:02.92]     Sorry.Analyzers.Tests.Sorry1001Tests.DoesNotWarnEmptyArgumentList [FAIL]
                                                                                                                                                X Sorry.Analyzers.Tests.Sorry1001Tests.DoesNotWarnEmptyArgumentList [1s 604ms]
  Error Message:
   RoslynTestKit.RoslynTestKitAnalyzerFailedException : Analyzer 'Sorry.Analyzers.Sorry1001WrongIndentation' threw the following exception:
'Exception occurred with following context:
Compilation: TestProject
SyntaxTree: TestDocument
SyntaxNode: void Method() { } [MethodDeclarationSyntax]@[66..125) (3,20)-(5,21)

System.NotImplementedException: The method or operation is not implemented.
   at Sorry.Analyzers.Sorry1001WrongIndentation.BaseMethodDeclarationAction(SyntaxNodeAnalysisContext context) in /Users/dmitryivanov/dev/sorry/src/Sorry.Analyzers/Sorry1001WrongIndentation.cs:line 40
   at Microsoft.CodeAnalysis.Diagnostics.AnalyzerExecutor.<>c__43`1.<ExecuteSyntaxNodeAction>b__43_0(ValueTuple`2 data)
   at Microsoft.CodeAnalysis.Diagnostics.AnalyzerExecutor.ExecuteAndCatchIfThrows_NoLock[TArg](DiagnosticAnalyzer analyzer, Action`1 analyze, TArg argument, Nullable`1 info)
-----
'.
  Stack Trace:
     at RoslynTestKit.AnalyzerTestFixture.NoException(String code) in /Users/dmitryivanov/dev/sorry/deps/RoslynTestKit/src/RoslynTestKit/AnalyzerTestFixture.cs:line 24
   at Sorry.Analyzers.Tests.Sorry1001Tests.DoesNotWarnEmptyArgumentList() in /Users/dmitryivanov/dev/sorry/tests/Sorry.Analyzers.Tests/Sorry1001Tests.cs:line 24
                                                                                                                                              
Test Run Failed.
```

Reply:
Hmm, so the former approach with `diagnostic.Descriptor.Description` seems to be a better solution because contains more information.

Reply:
Agree

Reply:
Thanks for collaboration